lib/parser/parser-utils.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Utils which parse components definition
+ */
+
+import * as path from 'path';
+import { ComponentsDefinitionV10X } from '../components-types-v1_0_x';
+import { DirectiveType, ParsedComponentsDefinition } from '../models';
+
+/**
+ * Returns information about directives
+ *
+ * @param content HTML content of the component
+ * @returns ParsedComponentsDefinition['components']['name']['directives']
+ */
+function parseDirectives(content: string) : ParsedComponentsDefinition['components']['name']['directives'] {
+    const regexTags = /<([a-z_\-]+).*?\s+doc-[a-z\d\-]+=['"]?[a-z\d\-]+['"]?.*?>/gmi;
```

Comment:
I wouldn't use regex to parse the html content, but rather an html parser like htmlparser2 and walk through the attributes.

Reply:
ok